src/utils/hooks/pipeline-assign.js

```diff
@@ -75,19 +79,19 @@ const assignPodToPipeline = async (message) => {
   // validate that the message contains input
   await validateRequest(message, 'PipelinePodRequest.v1.yaml');
 
-  const k8sApi = kc.makeApiClient(k8s.CoreV1Api);
-
   const { experimentId, input: { sandboxId, activityId, processName } } = message;
-  const namespace = `pipeline-${sandboxId}`;
 
-  logger.log(`Assigning pod to ${processName} pipeline for experiment ${experimentId} in sandbox ${sandboxId} for activity ${activityId}`);
-  // try to choose a free pod and assign it to the current pipeline
+  logger.log(`Trying to assign pod to ${processName} pipeline for experiment ${experimentId} in sandbox ${sandboxId} for activity ${activityId}`);
+
+
+  // remove pipeline pods already assigned to this experiment
+  await deleteExperimentPods(experimentId);
```

Comment:
can this be placed under the `try...catch` so exceptions are handled correctly? or should `deleteExperimentPods` create a proper exception that propagates up?